server/models/User.js

```diff
@@ -48,10 +59,40 @@ export default class User {
     return Promise.resolve(this.strip());
   }
 
+  async login() {
+    try {
+      const authQuery = `SELECT * FROM authentication WHERE username = '${this.username}'`;
+      const getAuth = await client.query(authQuery);
+      if (getAuth.rows.length === 0) return new Error('user not found');
+      this.username = getAuth.rows[0].username;
+      this.authId = getAuth.rows[0].id;
+      const correctPassword = await bcrypt.compare(this.password, getAuth.rows[0].password);
+      if (correctPassword) {
+        const userQuery = `SELECT * FROM users WHERE "authId"=${this.authId}`;
+        const getUser = await client.query(userQuery);
+        this.id = getUser.rows[0].id;
+        this.email = getUser.rows[0].email;
+        this.fullName = getUser.rows[0].fullName;
+        this.token = this.genToken();
```

Comment:
Yes, I'm sending the token back.
That is why I did `this.token = this.genToken();`
I send `this` back to the user (after the strip function removes it's password and authId)
@fidelisojeah